src/utils/metrics/metric_data.rs

```diff
@@ -0,0 +1,35 @@
+use crate::commands::send_metric::subcommands::{
+    CommonMetricArgs, FloatValueMetricArgs, IncrementArgs, SendMetricSubcommand, SetArgs,
+};
+
+pub(super) struct MetricData {
+    pub(super) key: String,
+    pub(super) value: f64,
+    pub(super) metric_type: char,
```

Comment:
I would suggest making an enum with all the different metric types, and storing the enum here. Then, at serialization, we would convert to a char